snapcraft/internal/repo/snaps.py

```diff
@@ -167,6 +167,8 @@ def is_classic(self) -> bool:
 
     def is_valid(self) -> bool:
         """Check if the snap is valid."""
+        if self.installed and self.get_local_snap_info()["channel"] == self.channel:
```

Comment:
the caveat here is that for this to work it has to be installed, which implies destructive-mode, correct?

Reply:
Or a preexisting environment?

Reply:
yes, I was thinking about the bug state after this lands as it alleviates the problem but does not completely fix it, that's all :-)